app/models/internal/TimeSeriesResult.java

```diff
@@ -69,15 +103,39 @@
      *
      * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
      */
-    interface QueryGroupBy {
-    }
+    @JsonTypeInfo(
+            use = JsonTypeInfo.Id.NAME,
+            property = "name"
+    )
+    @JsonSubTypes({
```

Comment:
These subtypes aren't as flexible as I'd like, but I can easily workaround this with a custom `Deserializer` class.

Ideally we would want to delegate each derived interface to the annotations on that interface, but Jackson doesn't resolve this recursively. This would be like a tree of `JsonTypeInfo`.